x-pack/plugins/lens/public/visualizations/partition/to_expression.ts

```diff
@@ -74,30 +78,25 @@ export const getSortedGroups = (
   return Array.from(new Set(originalOrder?.concat(layer[accessor] ?? [])));
 };
 
-const prepareDimension = (accessor: string) => {
-  const visdimension = buildExpressionFunction('visdimension', { accessor });
-  return buildExpression([visdimension]).toAst();
-};
+const prepareDimension = (accessor: string) =>
+  buildExpression([
+    buildExpressionFunction<ExpressionFunctionVisDimension>('visdimension', { accessor }),
+  ]).toAst();
 
 const generateCommonLabelsAstArgs: GenerateLabelsAstArguments = (state, attributes, layer) => {
-  const show = [!attributes.isPreview && layer.categoryDisplay !== CategoryDisplay.HIDE];
-  const position = layer.categoryDisplay !== CategoryDisplay.HIDE ? [layer.categoryDisplay] : [];
-  const values = [layer.numberDisplay !== NumberDisplay.HIDDEN];
-  const valuesFormat = layer.numberDisplay !== NumberDisplay.HIDDEN ? [layer.numberDisplay] : [];
-  const percentDecimals = [layer.percentDecimals ?? DEFAULT_PERCENT_DECIMALS];
+  const show = !attributes.isPreview && layer.categoryDisplay !== CategoryDisplay.HIDE;
+  const position =
+    layer.categoryDisplay !== CategoryDisplay.HIDE ? (layer.categoryDisplay as LabelPositions) : [];
+  const values = layer.numberDisplay !== NumberDisplay.HIDDEN;
+  const valuesFormat =
+    layer.numberDisplay !== NumberDisplay.HIDDEN ? (layer.numberDisplay as ValueFormats) : [];
```

Comment:
One last thought on this. I understand now why you're using an empty array for the default here. To make it more clear that it's just for the `buildExpressionFunction`'s sake, maybe you could do the defaulting when you invoke that function.

```ts
const valuesFormat = layer.numberDisplay !== NumberDisplay.HIDDEN ? (layer.numberDisplay as ValueFormats) : undefined;

...

  const partitionLabelsFn = buildExpressionFunction<PartitionLabelsExpressionFunctionDefinition>(
    'partitionLabels',
    { show, position, values, valuesFormat: valuesFormat ?? [], percentDecimals }
  );
```

No strong feeling here, just an idea I had 🤷‍♂️